localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -1020,12 +1021,21 @@ def update_global_table(data):
 # ---
 
 
-def is_index_query_valid(table_name: str, index_query_type: str) -> bool:
+def get_global_secondary_index(table_name, index_name):
     schema = SchemaExtractor.get_table_schema(table_name)
     for index in schema["Table"].get("GlobalSecondaryIndexes", []):
-        index_projection_type = index.get("Projection").get("ProjectionType")
-        if index_query_type == "ALL_ATTRIBUTES" and index_projection_type != "ALL":
-            return False
+        if index["IndexName"] == index_name:
+            return index
+
+
+def is_index_query_valid(query_data: dict) -> bool:
+    table_name = to_str(query_data["TableName"])
+    index_name = to_str(query_data["IndexName"])
+    index_query_type = query_data.get("Select")
+    index = get_global_secondary_index(table_name, index_name)
```

Comment:
Technically the `index` could be `None` here - can we raise an `Exception` and add a TODO comment: 

```
if not index:
    # TODO: add proper exception handling
    raise Exception("Index not found")
```

(we have another refactoring PR #5796 pending where we can then later pick this up.. 👍 ) Thanks